### PR TITLE
Course Run bug fixes

### PR DESCRIPTION
**Pull request by jlajoie**

### CODE REVIEW

File: src/components/EditCoursePage/CollapsibleCourseRunFields.jsx [hunk 1]
```diff
@@ -66,32 +66,35 @@ class CollapsibleCourseRunFields extends React.Component {
...
-            iconId={`collapsible-icon-${index}`}
+            key={`collapsible-run-${courseRun}`}
+            iconId={`collapsible-icon-${courseRun}`}
           >
             <Field
               name={`${courseRun}.start`}
               type="date"
               component={RenderInputTextField}
-              format={value => moment(value).format('YYYY-MM-DD')}
+              format={value => (value ? moment(value).format('YYYY-MM-DD') : '')}
```
Comment (bbaker6225) [inline src/components/EditCoursePage/CollapsibleCourseRunFields.jsx]: Maybe worth extracting to a helper function? Idk

File: src/components/EditCoursePage/CollapsibleCourseRunFields.jsx [hunk 2]
```diff
@@ -66,32 +66,35 @@ class CollapsibleCourseRunFields extends React.Component {
         {fields.map((courseRun, index) => (
           <Collapsible
             title={formatCourseRunTitle(courseRuns[index])}
-            iconId={`collapsible-icon-${index}`}
+            key={`collapsible-run-${courseRun}`}
+            iconId={`collapsible-icon-${courseRun}`}
```
Comment (bbaker6225) [inline src/components/EditCoursePage/CollapsibleCourseRunFields.jsx]: I think the `key` and `iconId` need a unique string, but they'll just get `[object Object]` on the end right now

File: src/components/EditCoursePage/EditCourseForm.jsx [hunk 3]
```diff
@@ -75,7 +75,7 @@ class EditCourseForm extends React.Component {
     const languageOptions = (courseRunOptions &&
       [{
         label: '----------', // Add a default null language for the dropdown
```
Comment (Dillon-Dumesnil) [inline src/components/EditCoursePage/EditCourseForm.jsx]: Nit: We should decide on consensus of how many dashes to put. Level type below uses two, here we have like 8

File: src/components/EditCoursePage/CollapsibleCourseRunFields.test.jsx [hunk 4]
```diff
@@ -3,6 +3,16 @@ import { shallow } from 'enzyme';
...
 
+/*
+*  Disable console errors for this test file so that we don't receive warnings
+*  about fields being an array rather than an object. This prop change is
+*  intentional as redux-form field arrays treat fields strangely:
+*  'The fields object is a "pseudo-array", in that it has many of the same
+*  properties and methods as a javascript Array, providing both reading and
+*  writing functionality.'
+*/
+jest.spyOn(global.console, 'error').mockImplementation(() => jest.fn());
```
Comment (Dillon-Dumesnil) [inline src/components/EditCoursePage/CollapsibleCourseRunFields.test.jsx]: 👀

File: src/components/EditCoursePage/CollapsibleCourseRunFields.jsx [hunk 5]
```diff
@@ -66,32 +66,35 @@ class CollapsibleCourseRunFields extends React.Component {
...
               component={RenderInputTextField}
-              format={value => moment(value).format('YYYY-MM-DD')}
+              format={value => (value ? moment(value).format('YYYY-MM-DD') : '')}
               label={
                 <React.Fragment>
                   Start date
                   <span className="required" aria-hidden>*</span>
                 </React.Fragment>
               }
+              placeholder="dd/mm/yyyy"
```
Comment (Dillon-Dumesnil) [inline src/components/EditCoursePage/CollapsibleCourseRunFields.jsx]: The placeholder is `dd/mm/yyyy`, but we format to `YYYY-MM-DD`. Is that just the way it should be viewed vs the way we need to send it to the API?

File: src/components/EditCoursePage/CollapsibleCourseRunFields.jsx (hunk 2 above)
Comment (jlajoie) [reply src/components/EditCoursePage/CollapsibleCourseRunFields.jsx]: courseRun comes as a string, string-thing-
Screen Shot 2019-04-01 at 2 37 37 PM
course_runs[0]

Comment (bbaker6225) [reply src/components/EditCoursePage/CollapsibleCourseRunFields.jsx]: Ok, cool. The snapshots deceived me

File: src/components/EditCoursePage/CollapsibleCourseRunFields.jsx (hunk 5 above)
Comment (jlajoie) [reply src/components/EditCoursePage/CollapsibleCourseRunFields.jsx]: Screen Shot 2019-04-01 at 3 45 50 PM
Internally it needs that date format, but the display is this

Comment (jlajoie) [reply src/components/EditCoursePage/CollapsibleCourseRunFields.jsx]: I do still need to fix it though :D

**Comment by mikedikan** [comment]
I think this PR may have (unintentionally?) removed Brandon's StaffList component from the collapsible course run.

File: src/components/EditCoursePage/CollapsibleCourseRunFields.jsx [hunk 6]
```diff
@@ -7,7 +7,6 @@ import { Collapsible } from '@edx/paragon';
 import RenderInputTextField from '../RenderInputTextField';
 import RenderSelectField from '../RenderSelectField';
 import TranscriptLanguage from './TranscriptLanguage';
-import StaffList from '../StaffList';
```
Comment (mikedikan) [inline src/components/EditCoursePage/CollapsibleCourseRunFields.jsx]: see here

File: src/components/EditCoursePage/CollapsibleCourseRunFields.jsx [hunk 7]
```diff
@@ -155,10 +159,6 @@ class CollapsibleCourseRunFields extends React.Component {
                 </React.Fragment>
               }
             />
-            <Field
```
Comment (mikedikan) [inline src/components/EditCoursePage/CollapsibleCourseRunFields.jsx]: and here

**Pull request merged**